python/ray/external_storage.py

```diff
@@ -423,11 +448,21 @@ def spill_objects(self, object_refs, owner_addresses) -> List[str]:
             return []
         from smart_open import open
 
+        # Choose the current uri by round robin order.
+        self._current_uri_index = (self._current_uri_index + 1) % len(self._uris)
+        uri = self._uris[self._current_uri_index]
+
         # Always use the first object ref as a key when fusioning objects.
         first_ref = object_refs[0]
         key = f"{self.prefix}-{first_ref.hex()}-multi-{len(object_refs)}"
-        url = f"{self.uri}/{key}"
-        with open(url, "wb", transport_params=self.transport_params) as file_like:
+        url = f"{uri}/{key}"
+        # with open(url, mode="wb", transport_params=self.transport_params) as file_like:
+        with open(
+            url,
+            mode="wb",
+            buffering=self._buffer_size,
```

Comment:
I believe this `buffering` parameter is not necessary. Reading the source https://github.com/RaRe-Technologies/smart_open/blob/develop/smart_open/smart_open_lib.py#L192, `buffering` only takes effect if the URI to open is a local file path, which is not the case here.

Could you test removing `buffering` here (and hence we can also get rid of `self._buffer_size`), to see if the buffer size still takes effect?